src/python/pants/backend/python/goals/run_pex_binary.py

```diff
@@ -100,12 +109,14 @@ def in_chroot(relpath: str) -> str:
         return os.path.join("{chroot}", relpath)
 
     complete_pex_env = pex_env.in_workspace()
-    args = complete_pex_env.create_argv(in_chroot(pex.name), python=pex.python)
+    args = complete_pex_env.create_argv(
+        in_chroot(entry_point_pex.name), python=entry_point_pex.python
+    )
 
     chrooted_source_roots = [in_chroot(sr) for sr in sources.source_roots]
     extra_env = {
-        **complete_pex_env.environment_dict(python_configured=pex.python is not None),
-        "PEX_PATH": in_chroot(local_dists.pex.name),
+        **complete_pex_env.environment_dict(python_configured=entry_point_pex.python is not None),
```

Comment:
You could just run requirements_pex with `"PEX_SCRIPT": f"{(entry_point.val or field_set.script.value).spec}"` but there is little harm in creating the additional entry point PEX just to hold the entry point.

Reply:
Oh, interesting! A lot of users have wondered why we create the `pytest_runner.pex` etc so many times. And we saw with one user that that step was taking ~3-4 seconds, we think due to sandbox creation time. It would be excellent to avoid creating that runner PEX.

Reply:
I agree it would probably be good to skip creating an un-needed PEX, but creating empty PEX should be as light a sandbox as they come. ~3-4 seconds to make that PEX makes no sense.

Reply:
Oh yes, that is much better. Will try that out.

Reply:
What about additional args?

Reply:
I think we still need the wrapper pex for those?

Reply:
```python
if self.emit_warnings.value_or_global_default(pex_binary_defaults) is False:
    args.append("--no-emit-warnings")
if self.ignore_errors.value is True:
    args.append("--ignore-errors")
if self.inherit_path.value is not None:
    args.append(f"--inherit-path={self.inherit_path.value}")
if self.shebang.value is not None:
    args.append(f"--python-shebang={self.shebang.value}")
if self.strip_env.value is False:
    args.append("--no-strip-pex-env")
if self._execution_mode is PexExecutionMode.VENV:
    args.extend(("--venv", "prepend"))
if self.include_tools.value is True:
    args.append("--include-tools")
```

The only ones of those you can't `PEX_*` control are `--python-shebang` which is irrelevant since we control the execution with `python user.pex` here and `--include-tools`. If tools were not included then `PEX_TOOLS=1 ./pants run ...` will not work (I'm not sure if we even de-hermeticize env vars for `./pants run` and pass those through) and `PEX_VENV` will not work since it requires pex tools are included.

So, you're right. Barely. Its just the `--include-tools` that blocks this.

Reply:
I'll punt on this for now and we can revisit if needed.